### PR TITLE
[Reviewer: Rob] Refactor out deployment resize function

### PR DESCRIPTION
**Pull request by eleanor-merry**
Rob, can you review this change to refactor the deployment resize commands. The main changes are to add a deployment utils module that holds the utility functions for the deployment class, and to make creating the local config a recipe in its own right

### CODE REVIEW

File: plugins/knife/knife-box-create.rb [hunk 1]
```diff
@@ -88,12 +88,14 @@ def run()
       end
       role = name_args.first
 
-      unless Clearwater::BoxManager.supported_roles.include? role
-        ui.fatal "#{role} is not a supported box role"
-        exit 1
+      if supported_boxes != []
+        Chef::Log.error "Adding boxes #{supported_boxes.to_s}"
```
Comment (rkd-msw) [inline plugins/knife/knife-box-create.rb]: These logs are a bit misleading.

File: plugins/knife/knife-clearwater-utils.rb [hunk 2]
```diff
@@ -103,5 +103,14 @@ def continue? (prompt = "Continue?")
       end
     end
 
+    # Expands out hashes of boxes, e.g. {:bono => 3} becomes:
+    # {{:role => "bono", :index => 1}, {:role => "bono", :index = 2}, etc...
+    def create_cluster(boxes)
```
Comment (rkd-msw) [inline plugins/knife/knife-clearwater-utils.rb]: Why is this in knife-clearwater-utils and not knife-deployment utils? (More generally, what are the criteria for choosing where to put new functions?)

File: plugins/knife/knife-deployment-utils.rb [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+        Chef::Log.info "Creating DNS records..."
+        dns_create = dns_class.new("-E #{config[:environment]}".split)
+        dns_create.config[:verbosity] = config[:verbosity]
+        Chef::Config[:verbosity] = config[:verbosity]
+        dns_create.run
+        status["DNS"][:status] = "Done"
+      end
+    end
+
+    def init_status(node_list, single_node_list)
```
Comment (rkd-msw) [inline plugins/knife/knife-deployment-utils.rb]: Could do with a comment explaining the difference between the parameters?

**Comment by rkd-msw** [comment]
The changes look sensible, but I'm increasingly struggling to understand what's going on with this code. I think we should improve the commenting of some of these methods (and maybe rename them - what should `init_status` even do?).

**Comment by eleanor-merry** [comment]
Agreed - I've gone through and generally updated comments/function names

**Pull request merged**